src/functions.c

```diff
@@ -370,6 +370,34 @@ void fcallCommandReadOnly(client *c) {
     fcallCommandGeneric(c, 1);
 }
 
+void functionsFlushCommand(client *c) {
+    if (c->argc > 3) {
+        addReplyErrorFormat(c,"wrong number of arguments for '%s' command or subcommand", c->cmd->name);
+        return;
+    }
+    int async = 0;
+    if (c->argc == 3 && !strcasecmp(c->argv[2]->ptr,"sync")) {
+        async = 0;
+    } else if (c->argc == 3 && !strcasecmp(c->argv[2]->ptr,"async")) {
+        async = 1;
+    } else if (c->argc == 2) {
+        async = server.lazyfree_lazy_user_flush ? 1 : 0;
+    } else {
+        addReplyError(c,"FUNCTION FLUSH only supports SYNC|ASYNC option");
+        return;
+    }
+
+    if (async) {
+        functionsCtx* old_f_ctx = functions_ctx;
+        functions_ctx = functionsCtxCreate();
+        freeFunctionsAsync(old_f_ctx);
+    } else {
+        functionsCtxClear(functions_ctx);
+    }
+    forceCommandPropagation(c, PROPAGATE_REPL | PROPAGATE_AOF);
```

Comment:
i now realize that since functions are saved to the rdb, we need to do server.dirty++ (to trigger a periodic save later).
so this comes in place of forceCommandPropagation, and probably applies in the other FUNCTION sub-commands.

Reply:
@oranagra right, fixed it on this PR  for `FUNCTION FLUSH` and will make a seperate PR to fix it for `FUNCTION CREATE` and `FUNCTION DELETE`

Reply:
PR for `FUNCTION CREATE` and `FUNCTION DELETE`: https://github.com/redis/redis/pull/9945